R/profilePlot.R

```diff
@@ -112,209 +105,322 @@ profilePlot <- function(contrastDF,
         if (!is.null(referenceLine)) {
-            referenceLine <- rgbaConversion(referenceLine, alpha = alpha)
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color = referenceLine,
-                                                width = refLineThickness,
-                                                y     = 0)
+            referenceLine <- .rgbaConversion(referenceLine, alpha = transparency)
+            decorations   <- .getCxPlotDecorations(decorations = decorations,
+                                                   color = referenceLine,
+                                                   width = refLineThickness,
+                                                   y     = 0)
         }
-
         if (!is.null(foldChangeLines)) {
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color       = symbolFill[which(groupNames == "Increased")],
-                                                width       = refLineThickness,
-                                                y           = foldChangeLines)
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color       = symbolFill[which(groupNames == "Decreased")],
-                                                width       = refLineThickness,
-                                                y           = -1*foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[2],
+                                                 width       = refLineThickness,
+                                                 y           = foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[1],
+                                                 width       = refLineThickness,
+                                                 y           = -1 * foldChangeLines)
+        }
+        events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
+                                                if (o != null && o != false) {
+                                                  if (o.y != null &&
+                                                      o.y.data != null &&
+                                                      o.y.smps != null) {
+                                                      info = '<b>' + o.y.vars[0]  + '</b>' + '<br/>' +
+                                                             '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
+                                                             '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1] ;
+                                                      if (o.z != null && o.z['rgd_symbol'] != null) {
```

Comment:
this will result in a bug. rgd_symbol is hardcoded. Not all geneSymcols will have the same column name.

Reply:
done